### ISSUE TITLE
The image is dull

### ISSUE THREAD

**Issue by stla** [cj-holmes/isocuboids#1]
Hello,

This looks like an amazing package. I've just tried it, starting with this image:

![Klein_t0](https://github.com/cj-holmes/isocuboids/assets/4466543/7dd56dcc-0dce-4a2a-bf02-4de251d8ffe3)

As you can see, the green color is rather dark. But I did an animation and on the "big" image, the one in front of you, the green is less dark, the image becomes dull. Do you know why and how to avoid that ? I hope my English is clear.

![Klein_isocuboids](https://github.com/cj-holmes/isocuboids/assets/4466543/b4415c56-44f1-48ec-acb0-4bd8d7d0acf6)

**Comment by cj-holmes** [comment]
Could you provide a reproducible example of the code? Im not sure why the colour is changing with different angles

**Comment by stla** [comment]
Yes, sure. Here is the code. The original image is the I posted, named `Klein_t0.png`.

Is there some lighting in the scene?

```r
library(magick)
library(isocuboids)
library(ggplot2)

t_ <- seq(0, 1, length.out = 20L)
for(i in seq_along(t_)) {
  print(i)
  t <- t_[i]
  ci <- cuboid_image(
    "Klein_t0.png", res = 512, show_axes = FALSE,
    a1 = t*45, a2 = t*35 + (1-t)*90
  ) + theme(
    panel.background = element_blank(),
    axis.text = element_blank(),
    axis.title = element_blank(),
    axis.ticks = element_blank()
  )
  ggsave(
    sprintf("zzpic%03d.png", i), 
    ci, width=512, height=512, units = "px", dpi = 600
  )
}

library(gifski)
pngs <- Sys.glob("zzpic*.png")
gifski(
  c(pngs, rev(pngs)),
  "Klein_isocuboids.gif",
  width = 512, height = 512,
  delay = 1/4
)

file.remove(pngs)
```

**Comment by cj-holmes** [comment]
Hmmm yeah it looks like the slight grey lines around the cuboids make the image look darker at high angle.
I'm really not sure if this is a rendering problem or maybe a problem due to the cuboboids not aligning to a very high precision?

**Comment by cj-holmes** [comment]
Do you get better results with this? 
It uses ```theme_void()``` and specifies a larger image output in inches (5x5)

```
library(magick)
library(isocuboids)
library(ggplot2)

t_ <- seq(0, 1, length.out = 20L)
for(i in seq_along(t_)) {
    print(i)
    t <- t_[i]
    ci <- cuboid_image(
        "Klein_t0.png", res = 512, show_axes = FALSE,
        a1 = t*45, a2 = t*35 + (1-t)*90
    ) + theme_void()
    
    ggsave(
        sprintf("zzpic%03d.png", i), 
        ci, width=5, height=5, units = "in", dpi = 600
    )
}

library(gifski)
pngs <- Sys.glob("zzpic*.png")
gifski(
    c(pngs, rev(pngs)),
    "Klein_isocuboids.gif",
    width = 512, height = 512,
    delay = 1/4
)

file.remove(pngs)
```

**Comment by stla** [comment]
I observed that the plot is correct in the RStudio viewer. So it seems the issue is due to `ggsave`. 

I'm going to read your latest message now.

**Comment by cj-holmes** [comment]
Not sure its a problem with ```ggsave``` but maybe the resolution and size of the image being saved. I'm not sure if ```theme_void()``` makes a difference!

**Comment by stla** [comment]
Not tried yet, please wait a minute.

Usually when I save a plot, I [firstly save it to SVG then I convert it to PNG](https://laustep.github.io/stlahblog/posts/SVG2PNG.html). This gives a highly better quality.

I've just tried:

```r
svg("x.svg")
cuboid_image(
  "Klein_t0.png", res = 512, show_axes = FALSE,
  a1 = 0, a2 = 90
) + theme(
  panel.background = element_blank(),
  axis.text = element_blank(),
  axis.title = element_blank(),
  axis.ticks = element_blank()
)
dev.off()

rsvg::rsvg_png("x.svg", "ztest.png", width = 512, height = 512)
```
But `rsvg` has crashed the session. The SVG has 55Mb and it is not nice...

But again, the plot is perfect in the RStudio viewer.

**Comment by stla** [comment]
Here is a screenshot of the SVG:

![kleinisoc](https://github.com/cj-holmes/isocuboids/assets/4466543/90b94b37-0758-45be-b5dd-dba05dd48d7c)

**Comment by stla** [comment]
Indeed, it looks better with `theme_void`. I didn't try the GIF yet.

![zzpic000](https://github.com/cj-holmes/isocuboids/assets/4466543/c0cdd44e-b69e-467e-8dcd-0d63f9ee0397)

**Comment by stla** [comment]
With `theme_void` and 5 inches:

![zzpic001](https://github.com/cj-holmes/isocuboids/assets/4466543/a17f5925-4423-43d5-8cd8-beec0993213d)

Nice. Can't upload the GIF, too big.

**Comment by stla** [comment]
Ah no, the GIF is not too big. And it is fine! Looks like you found the solution.

![Klein_isocuboids](https://github.com/cj-holmes/isocuboids/assets/4466543/0448cf8a-2417-47a4-bcd7-8f72d8f4074b)

**Comment by cj-holmes** [comment]
Great glad its working
